test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataInCookiesTest.cs

```diff
@@ -135,5 +135,35 @@ public TempDataInCookiesTest(MvcTestFixture<BasicWebSite.StartupWithCookieTempDa
             Assert.NotNull(setCookieHeader.Expires);
             Assert.True(setCookieHeader.Expires < DateTimeOffset.Now); // expired cookie
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task CookieTempDataProviderCookie_SetsSecureAttributeOnCookie_OnlyIfRequestIsSecure(bool secureRequest)
```

Comment:
I think it's critical to have a functional test for this because the question is what do actual clients do. Whether it can be merged with some other test, I don't know.